cachelib/allocator/CacheAllocator-inl.h

```diff
@@ -1485,6 +1548,71 @@ bool CacheAllocator<CacheTrait>::moveRegularItem(Item& oldItem,
   return true;
 }
 
+template <typename CacheTrait>
+bool CacheAllocator<CacheTrait>::moveChainedItemWithSync(ChainedItem& oldItem,
+                                                 WriteHandle& newItemHdl,
+                                                 WriteHandle& parentHandle) {
+  util::LatencyTracker tracker{stats_.moveChainedLatency_};
+
+  auto& expectedParent = oldItem.getParentItem(compressor_);
+  const auto parentKey = expectedParent.getKey();
+  auto l = chainedItemLocks_.tryLockExclusive(parentKey);
```

Comment:
Do we still need `chaineItemLocks_` if we use `moving` bit for synchronization?

Reply:
If I remember correctly, we still want to use chainedItemLocks_ to allow modifying the chain even if there is someone holding Handle to the parent item.

Is it still true for this implementation?